### DISCUSSION TITLE
It's timeout while return a response with a large dictionary

### DISCUSSION THREAD

**Discussion by lanlanlaura** [fastapi/fastapi#11801] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import os

path = os.path.dirname(__file__)
file_path = os.path.join(path, "test.json")
file = json.load(open(file_path, 'r', encoding='utf-8'))

app = FastAPI()
@app.get("/read")
def read_dic():
    return file
```

### Description

The test.json file size is around 2M, I convert it to dictionary,  It's timeout, while FastAPI return a response. Thank you very much for your help.

### Operating System

Windows

### FastAPI Version

0.110.1

### Pydantic Version

2.6.4

### Python Version

3.8.10

**Comment by AIdjis** [comment, marked as answer]
Hi , i found the solution to your problem, first you need to install orjson because it load the josn file much faster compered to json. 
1- install orjson:
```console
 pip install orjson 
```
2- use this code :

```python  
from fastapi import FastAPI
import os
import json
import orjson

@app.get("/read-json")
async def read_json_file():
    path = os.path.dirname(__file__)
    file_path = os.path.join(path, "file.json")
    with open(file_path, 'rb') as file:
        content = file.read()
        data = dict(orjson.loads(content))
        return data
```
Then send the request to the endpoint using Postman. I have already tried it with a file larger than 2MB, and the response was very quick.
I hope I answered your question.

**Reply by lanlanlaura** [reply]
Hi, Djemmal Islam,
Many thanks for your answer. I try this code and go to "http://127.0.0.1:8001/docs" to return data, as the data is huge, the page is block and I need to wait around 1min to get the data. I'm a beginner of FastAPI, This is truly a very tricky question for me.  Do you have a suggestion for it? Thank you again. 
![image](https://github.com/tiangolo/fastapi/assets/42463884/a9db3088-06e7-40c5-bbd9-0e160526e1f2)

**Reply by YuriiMotov** [reply]
Try removing `async`:
`def read_json_file():` instead of `async def read_json_file():`

The code of this endpoint is sync and if the file is big it will block event loop

**Reply by PhysicallyActive** [reply]
I tried it out using both Swagger and Curl (and also the two different implementations), and I'm fairly sure the issue lies with Swagger.
That might be why it works for @AIdjis (since Postman was used for the verification).

If you try to curl your endpoint:
```bash
curl -X GET http://127.0.0.1:8001/read-json -H "content-type: application/json" > tmp.json
```

and compare the result, they're identical.

**Reply by jgould22** [reply]
The problem is you are using swagger and when it gets a response back it applies syntax highlighting and other processing to the 2mb of json which can take a lot of time in a browser.

https://fastapi.tiangolo.com/ru/how-to/configure-swagger-ui/#disable-syntax-highlighting

**Reply by lanlanlaura** [reply]
Many thanks, all of you. I know why the problem comes up. Thank you again for your help!
